src/mlpack/methods/boostmetric/boostmetric.hpp

```diff
@@ -0,0 +1,150 @@
+/**
+ * @file boostmetric.hpp
+ * @author Manish Kumar
+ *
+ * Declaration of BoostMetric class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_BOOSTMETRIC_BOOSTMETRIC_HPP
+#define MLPACK_METHODS_BOOSTMETRIC_BOOSTMETRIC_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/metrics/lmetric.hpp>
+#include <mlpack/methods/lmnn/constraints.hpp>
+
+namespace mlpack {
+namespace boostmetric /** BoostMetric Distance Learning. */ {
+
+/**
+ * An implementation of BoostMetric Distance Learning technique.
+ * The method seeks to improve clustering & classification algorithms on
+ * a dataset by transforming the dataset representation in a more convenient
+ * form for them. It introduces the concept of target neighbors and impostors,
+ * focusing on the idea that the distance between impostors and the perimeters
+ * established by target neighbors should be large and that between target
+ * neighbors and data point should be small. It uses an adaptation of
+ * coordinate descent optimization as a boosting technique.
+ *
+ * For more details, see the following published paper:
+ *
+ * @code
+ * @incollection{NIPS2009_3658,
+ *   title = {Positive Semidefinite Metric Learning with Boosting},
+ *   author = {Shen, Chunhua and Junae Kim and Wang, Lei and Anton Hengel},
+ *   booktitle = {Advances in Neural Information Processing Systems 22},
+ *   editor = {Y. Bengio and D. Schuurmans and J. D. Lafferty and C. K. I.
+ *       Williams and A. Culotta},
+ *   pages = {1651--1659},
+ *   year = {2009},
+ *   publisher = {Curran Associates, Inc.},
+ *   url = {http://papers.nips.cc/paper/
+ *       3658-positive-semidefinite-metric-learning-with-boosting.pdf}
+ * }
+ * @endcode
+ *
+ * @tparam MetricType The type of metric to use for computation.
+ */
+template<typename MetricType = metric::SquaredEuclideanDistance>
+class BOOSTMETRIC
```

Comment:
Seems to me like we should call it `BoostMetric`.

Reply:
Sure